google-cloud-storage/src/main/java/com/google/cloud/storage/multipartupload/model/UploadResponseParser.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.storage.multipartupload.model;
+
+import com.google.api.client.http.HttpResponse;
+import com.google.api.core.BetaApi;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/** A utility class to parse {@link HttpResponse} and create a {@link UploadPartResponse}. */
+@BetaApi
+public final class UploadResponseParser {
```

Comment:
This should be moved to `com.google.cloud.storage` and made package private. We don't expose HttpResponses in our public api. A customer should never need to call this code.
```suggestion
final class UploadResponseParser {
```

Reply:
I kept it in multipartupload.model as its just specific to mpu, but I agree it makes sense to keep with the rest, specially when it allows us to make it package private.